luigi/contrib/snowflake.py

```diff
@@ -0,0 +1,411 @@
+import abc
+import logging
+
+import luigi
+from luigi.contrib import rdbms
+from luigi.contrib import redshift
+from luigi.contrib import postgres
+
+logger = logging.getLogger('luigi-interface')
+
+# Snowflake's library do not have constant for their error code.
+# We define our own
+SNOWFLAKE_TABLE_EXISTS_ERROR = 2002
+SNOWFLAKE_GENERIC_PROGRAMMING_ERROR = 2003
+
+try:
+    import snowflake
+    import snowflake.connector
+except ImportError:
+    logger.warning("Loading snowflake module without snowflake-connector-python installed. "
+                   "Will crash at runtime if snowflake functionality is used.")
+
+
+class _SettingsMixins():
```

Comment:
do we need to inherit from `object`?

Reply:
From what I understand `object` would only be required for Python 2.X while it's not necessary for Python 3.X unless we're trying to be python version agnostic here. More information around that [concept here](https://stackoverflow.com/a/45062077).

Reply:
for now, Luigi still supports (and tests) 2.7